pkg/operator/controller/status/controller.go

```diff
@@ -421,8 +426,8 @@ func computeOperatorProgressingCondition(ingresscontrollers []operatorv1.Ingress
 		}
 	}
 
-	if !allIngressesAvailable {
-		messages = append(messages, "Not all ingress controllers are available.")
+	if allIngressesAvailable != nil {
+		messages = append(messages, allIngressesAvailable.Error())
```

Comment:
possibly worth packing the `Reason` along as structured information, with some sort off aggregation in the event that there are multiple underlying reasons among different ingress controllers, to avoid the generic `progressingCondition.Reason = "Reconciling"` below, and make it easier to see what sort of failure modes we're having in Telemetry and other systems that can process `reason` on ClusterOperator conditions.  But I haven't bitten that off in this pull request.